### PR TITLE
add server side empty renegotiation_info support

### PR DESCRIPTION
**Pull request by cconlon**
This PR enables a wolfSSL server to send an empty renegotiation_info extension in response to a client probing for secure renegotiation support as per RFC 5746.

When enabled, the server will respond with an empty "renegotiation_info" extension in the ServerHello message when a ClientHello is received that includes either the presence of the TLS_EMPTY_RENEGOTIATION_INFO_SCSV cipher suite, or an empty renegotiation_info TLS extension.

To enable, define **HAVE_SERVER_RENEGOTIATION_INFO** and TLS extensions (HAVE_TLS_EXTENSIONS, or "--enable-tlsx"):

```
$ cd wolfssl
$ ./configure --enable-tlsx CFLAGS="-DHAVE_SERVER_RENEGOTIATION_INFO"
$ make
```

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -17820,6 +17820,28 @@ int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
...
+        int i;
+
+        if (suites == NULL) {
+            WOLFSSL_MSG("Suites pointer error");
+            return SUITES_ERROR;
+        }
+
+        for (i = 0; i < suites->suiteSz; i += 2) {
+            if (suites->suites[i]   == first &&
+                suites->suites[i+1] == second )
```
Comment (toddouska) [inline src/internal.c]: i+1 may exceed suiteSz

File: src/internal.c [hunk 2]
```diff
@@ -17820,6 +17820,28 @@ int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
...
+    static int FindSuite(Suites* suites, byte first, byte second)
+    {
+        int i;
+
+        if (suites == NULL) {
+            WOLFSSL_MSG("Suites pointer error");
+            return SUITES_ERROR;
+        }
+
+        for (i = 0; i < suites->suiteSz; i += 2) {
```
Comment (toddouska) [inline src/internal.c]: SUITE_LEN is probably better than '2'

Comment (cconlon) [inline src/internal.c]: Good catch, will switch to SUITE_LEN

File: src/internal.c (hunk 1 above)
Comment (cconlon) [inline src/internal.c]: Switched "suites->suiteSz" to "suites->suiteSz-1"

File: src/internal.c [hunk 3]
```diff
@@ -17820,6 +17820,28 @@ int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
...
     }
 
+#ifdef HAVE_SERVER_RENEGOTIATION_INFO
+
+    /* search suites for specific one, idx on success, negative on error */
+    static int FindSuite(Suites* suites, byte first, byte second)
+    {
+        int i;
+
+        if (suites == NULL) {
```
Comment (toddouska) [inline src/internal.c]: Needs an || check for suiteSz == 0 as well.  Because if suiteSz is 0 then suiteSz - 1 on 17,835 will be a big number (word16).

Comment (cconlon) [inline src/internal.c]: Fixing now, adjusted commit will be in shortly.

**Pull request merged**